### DISCUSSION TITLE
Detecting security problems in a Fortran code

### DISCUSSION THREAD

**Discussion by vmagnin** [marshallward/flint#15] [Ideas]
It would be interesting if such a tool could have an option to detect security issues, such as those discussed here: 
https://fortran-lang.discourse.group/t/what-about-security-issues-in-fortran/102

Just printing the lines containing the most problematic instructions, like `execute_command_line()` would be a first step.

**Comment by marshallward** [comment]
Thanks, I agree a list of potential security holes would be a good idea.  Both `execute_command_line` and `system` are both major causes of worry.  Stack overflow and runtime access to the object code also come to mind.

I've been following that thread, interesting to see how Fortran fits into the subject.  So much of Fortran operates within its own virtual machine that it can make us forget the potential problems.
